service/src/main/java/com/derpgroup/derpwizard/manager/DerpWizardManager.java

```diff
@@ -32,43 +32,100 @@
     serviceOutput.getVoiceOutput().setSsmltext("Hi. This is DerpWizard.");
     serviceOutput.getVoiceOutput().setPlaintext("Hi. This is DerpWizard.");
     serviceOutput.setConversationEnded(true);
   }
 
-  @Override
-  protected void doGoodbyeRequest(VoiceInput voiceInput, ServiceOutput serviceOutput) {
+  protected void doGoodbyeRequest(ServiceInput voiceInput, ServiceOutput serviceOutput) {
     serviceOutput.getVoiceOutput().setSsmltext("Goodbye!");
     serviceOutput.getVoiceOutput().setPlaintext("Goodbye!");
     serviceOutput.setConversationEnded(true);
   }
 
-  @Override
-  protected void doConversationRequest(VoiceInput voiceInput, ServiceOutput serviceOutput) {
-    serviceOutput.getVoiceOutput().setSsmltext("I'd love to help, but I'm not programmed to have conversations yet.");
-    serviceOutput.getVoiceOutput().setPlaintext("I'd love to help, but I'm not programmed to have conversations yet.");
+  protected void doCancelRequest(ServiceInput voiceInput, ServiceOutput serviceOutput) {
+    serviceOutput.getVoiceOutput().setSsmltext("Cancelling");
+    serviceOutput.getVoiceOutput().setPlaintext("Cancelling");
     serviceOutput.setConversationEnded(true);
   }
 
-  @Override
-  protected void doCancelRequest(VoiceInput voiceInput, ServiceOutput serviceOutput) {
+  protected void doStopRequest(ServiceInput voiceInput, ServiceOutput serviceOutput) {
+    serviceOutput.getVoiceOutput().setSsmltext("Stopping");
+    serviceOutput.getVoiceOutput().setPlaintext("Stopping");
     serviceOutput.setConversationEnded(true);
   }
 
-  @Override
-  protected void doStopRequest(VoiceInput voiceInput, ServiceOutput serviceOutput) {
+  protected void doRepeatRequest(ServiceInput voiceInput, ServiceOutput serviceOutput) {
+    serviceOutput.getVoiceOutput().setSsmltext("Repeating");
+    serviceOutput.getVoiceOutput().setPlaintext("Repeating");
     serviceOutput.setConversationEnded(true);
   }
+
+  protected void doYesRequest(ServiceInput voiceInput, ServiceOutput serviceOutput) {
+    serviceOutput.getVoiceOutput().setSsmltext("Yes");
+    serviceOutput.getVoiceOutput().setPlaintext("Yes");
+    serviceOutput.setConversationEnded(true);
+  }
+
+  protected void doNoRequest(ServiceInput voiceInput, ServiceOutput serviceOutput) {
+    serviceOutput.getVoiceOutput().setSsmltext("No");
+    serviceOutput.getVoiceOutput().setPlaintext("No");
+    serviceOutput.setConversationEnded(true);
+  }
+
+  /**
+   * An example primary entry point into the service.
+   * At this point the Resource classes should have mapped any device-specific requests
+   * into standard ServiceInput/ServiceOutput POJOs. As well as mapped any device-specific
+   * requests into service understandable subjects.
+   * @param serviceInput
+   * @param serviceOutput
+   */
+  public void handleRequest(ServiceInput serviceInput, ServiceOutput serviceOutput){
+    switch(serviceInput.getSubject()){
```

Comment:
Now that we don't have the weird intermediate stage, should each service just provide an enum to switch on?